src/main/java/ch/njol/skript/command/ScriptCommand.java

```diff
@@ -459,21 +460,27 @@ public void registerHelp() {
 	}
 
 	public void unregisterHelp() {
-		Bukkit.getHelpMap().getHelpTopics().removeAll(helps);
-		final HelpTopic aliases = Bukkit.getHelpMap().getHelpTopic("Aliases");
-		if (aliases != null && aliases instanceof IndexHelpTopic) {
-			try {
-				final Field topics = IndexHelpTopic.class.getDeclaredField("allTopics");
-				topics.setAccessible(true);
-				@SuppressWarnings("unchecked")
-				final ArrayList<HelpTopic> as = new ArrayList<>((Collection<HelpTopic>) topics.get(aliases));
-				as.removeAll(helps);
-				topics.set(aliases, as);
-			} catch (final Exception e) {
-				Skript.outdatedError(e);//, "error unregistering aliases for /" + getName());
+		new Task(Skript.getInstance(), 1) {
```

Comment:
```suggestion
		new Task(Skript.getInstance(), 0) {
```
Do you think Task.CallSync would be more appropriate? Ideally this thread shouldn't proceed until the help is removed.

Reply:
Don't need it to return a value

Reply:
The whole code block is just getting moved into a sync calling. There is no other changes from the existing behaviour.

Reply:
it's specifically that with async/parallel loading enabled, i do not believe ScriptLoader forces unloading onto another thread (though perhaps it should in the future!). i don't know how it could be on another thread unless you're executing the command (or unload effect) from another thread. is that the case?

Reply:
Ok, i've swapped to callSync, but to counter the bottle neck waiting for each structure to sync, I used parallel streams if the script loader is set to parallel.

Reply:
that seems reasonable